internal/handlers/common/update.go

```diff
@@ -118,5 +123,78 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 		}
 	}
 
-	return true, nil
+	return changed, nil
+}
+
+// ValidateUpdateOperators validates update statement.
+func ValidateUpdateOperators(update *types.Document) error {
+	var err error
+	set, err := getUpdateOperatorDocument("$set", update)
+	if err != nil {
+		return err
+	}
+	_, err = getUpdateOperatorDocument("$setOnInsert", update)
+	if err != nil {
+		return err
+	}
+
+	inc, err := getUpdateOperatorDocument("$inc", update)
+	if err != nil {
+		return err
+	}
+
+	if set != nil && inc != nil {
+		for _, setKey := range set.Keys() {
+			if inc.Has(setKey) {
+				return NewWriteErrorMsg(
+					ErrConflictingUpdateOperators,
+					fmt.Sprintf("Updating the path '%[1]s' would create a conflict at '%[1]s'", setKey),
+				)
+			}
+		}
+	}
+
+	for _, updateOp := range update.Keys() {
+		switch updateOp {
+		case "$set",
+			"$setOnInsert",
+			"$inc":
```

Comment:
Please put all cases on the same line, or use 3 separate cases

Reply:
oops, I wish I had linter here. fixing.

Reply:
done